src/main/java/io/cdap/plugin/WindowAggregation.java

```diff
@@ -259,186 +250,6 @@ private String generateFrameDescription(WindowAggregationConfig config) {
-        return functions.callUDF("percentile", functions.col(data.getFieldName()), lit);
-      }
-      case DISCRETE_PERCENTILE: {
-        return functions.callUDF("DISCRETE_PERCENTILE", functions.col(data.getFieldName()));
-      }
-      case LEAD: {
-        checkFunctionArguments(args, 1, "LEAD");
-        return functions.lead(data.getFieldName(), Integer.parseInt(args[0]));
-      }
-      case LAG:
-        checkFunctionArguments(args, 1, "LAG");
-        return functions.lag(data.getFieldName(), Integer.parseInt(args[0]));
-      case FIRST: {
-        boolean ignoreNull = data.isIgnoreNull();
-        if (args.length != 0) {
-          ignoreNull = ignoreNull || Boolean.parseBoolean(args[0]);
-        }
-        return functions.first(data.getFieldName(), ignoreNull);
-      }
-      case LAST: {
-        boolean ignoreNull = data.isIgnoreNull();
-        if (args.length != 0) {
-          ignoreNull = ignoreNull || Boolean.parseBoolean(args[0]);
-        }
-        return functions.last(data.getFieldName(), ignoreNull);
-      }
-      case CUMULATIVE_DISTRIBUTION:
-        return functions.cume_dist();
-      case ACCUMULATE:
-        return functions.sum(data.getFieldName());
-      default:
-        throw new InvalidParameterException("Can not find function " + data.getFieldName());
-    }
-  }
-
-  private void checkFunctionArguments(String[] args, int minimumArguments, String function) {
-    if (args.length < minimumArguments) {
-      throw new InvalidParameterException(
-        String.format("%s must have at least %s arguments", function, minimumArguments));
-    }
-  }
-
-  private Schema schemaForInputField(Schema inputSchema, String inputFieldName) {
-    if (inputSchema == null) {
-      return null;
-    }
-
-    Schema.Field schemaField = inputSchema.getField(inputFieldName);
-    if (schemaField == null) {
-      return null;
-    }
-
-    return schemaField.getSchema();
-  }
-
-  /**
-   * Function to map from {@link Row} to {@link StructuredRecord}.
-   */
-  public static final class RowToRecord implements
-    Function<Row, StructuredRecord> {
-
-    private final Schema outputSchema;
-
-    public RowToRecord(Schema outputSchema) {
-      this.outputSchema = outputSchema;
-    }
-
-    @Override
-    public StructuredRecord call(Row row) throws Exception {
-
-      Schema rowSchema = DataFrames.toSchema(row.schema());
-      StructuredRecord structuredRecord = DataFrames.fromRow(row, rowSchema);
-
-      if (outputSchema == null || outputSchema.getFields() == null) {
-        throw new Exception("Invalid output schema");
-      }
-
-      StructuredRecord.Builder builder = StructuredRecord.builder(outputSchema);
-      for (Schema.Field field : outputSchema.getFields()) {
-        builder.set(field.getName(), structuredRecord.get(field.getName()));
-      }
-      return builder.build();
-    }
+    return Utils.transform(sparkExecutionPluginContext, javaRDD, config, inputSchema, outputSchema);
```

Comment:
The Utils.transform method seems to be only used once, here. Can we just keep that logic here? Generally not good to have a lot of logic in static util classes unless absolutely necessary.

Reply:
We faced the same issue as here: https://cdap.atlassian.net/browse/CDAP-15912, because in this method we use a lot of sparks classes and we where forced to move to a different class.